src/platform/Zephyr/PlatformManagerImpl.cpp

```diff
@@ -61,6 +63,35 @@ static int app_entropy_source(void * data, unsigned char * output, size_t len, s
 }
 #endif // !CONFIG_NORDIC_SECURITY_BACKEND
 
+void PlatformManagerImpl::OperationalHoursSavingTimerEventHandler(k_timer * timer)
+{
+    PlatformMgr().ScheduleWork(UpdateOperationalHours);
+}
+
+void PlatformManagerImpl::UpdateOperationalHours(intptr_t arg)
+{
+    System::Clock::Timestamp currentTime = System::SystemClock().GetMonotonicTimestamp();
+
+    if (currentTime < sInstance.mLastOperationalHoursSaveTime)
+        return;
+
+    uint32_t totalOperationalHours = 0;
+    if (ConfigurationMgr().GetTotalOperationalHours(totalOperationalHours) == CHIP_NO_ERROR)
+    {
+        ConfigurationMgr().StoreTotalOperationalHours(
+            totalOperationalHours +
+            static_cast<uint32_t>(
+                std::chrono::duration_cast<System::Clock::Seconds64>(currentTime - sInstance.mLastOperationalHoursSaveTime)
```

Comment:
The problem with this solution is that it only records full hours and loses information about minutes and seconds (in case the timer was delayed etc). I would rather recommend to keep something like `mOperationalHoursSinceReboot` instead of `mLastOperationalHoursSaveTime` and calculate the `TotalOperationalHours` attribute as `ConfigurationMgr().GetTotalOperationalHours() - mOperationalHoursSinceReboot + GetUpTime()`. That will produce the most precise metrics. Also, it's worth checking whether the number of full hours increased before storing it in flash to assure the mechanism is correct even if one task is delayed and another one arrives on time, hence the interval between them is less than 1 hour.

Reply:
Ok, now it should work, as suggested